### PR TITLE
Adds small guide on how to start the LS in IntelliJ (to debug it)

### PR DESCRIPTION
**Pull request by soerendomroes**
As requested [here](https://github.com/lf-lang/lingua-franca/pull/2260#pullrequestreview-2006893265), I added a small guide on how to use the proposed IntelliJ run configuration.

### CODE REVIEW

File: docs/developer/developer-intellij-setup.mdx [hunk 1]
```diff
@@ -69,6 +69,22 @@ While debugging, you can run code step-by-step by using the debugger tools.
 
 By clicking the play button, `lfc` will be invoked, and if compilation is successful, its output can be found, relative to package root of the file under compilation, in `bin` if the target is a compiled language (e.g., C) or in `src-gen` if the target is an interpreted language (e.g., TypeScript).  For the `HelloWorld.lf` example above, the binary can be found in `test/Cpp/bin/HelloWorld` and can be executed in the terminal.
 
+## Starting and Debugging the Language Server
+
+There should a predefined run configuration called <kbd>LanguageDiagramServer</kbd> available, which executes the main method the `LanguageDiagramServer` class, which, as a result, starts the language server.
```
Comment (edwardalee) [inline docs/developer/developer-intellij-setup.mdx:74]: Is this actually predefined? Or does the user have to define it?

**Review by edwardalee — APPROVED** [review]
LGTM.  One small question inserted.

File: docs/developer/developer-intellij-setup.mdx (hunk 1 above)
Comment (soerendomroes) [reply docs/developer/developer-intellij-setup.mdx:74]: It is predefined once [this PR](https://github.com/lf-lang/lf-lang.github.io/pull/253) gets merged. I cloned the reository once more and checked out the sdo/debug-ls branch and the run configuration does just appear as the currently selected run configuration.

**Comment by soerendomroes** [comment]
> Does this work with the standard VS Code extension installed, or do we need a modified one? I am wondering what happens if we start the external language server and the extension tries to start its internal one.

Yes, as long as you start the run configuration to connect to the socket. [This](https://github.com/lf-lang/vscode-lingua-franca/blob/38515b4aa0de68383a88a0d4daf56cdb8c5b9b4c/.vscode/launch.json#L5-L13) run configuration is for the internal language server and [this](https://github.com/lf-lang/vscode-lingua-franca/blob/38515b4aa0de68383a88a0d4daf56cdb8c5b9b4c/.vscode/launch.json#L16-L39) run configuration is for an external one that is started on the socket 7670. The second one does however assume that the klighd-vscode extension is directly next to the vscode-lingua-franca repository. This is, however, not necessary and we be easily modified.

**Comment by soerendomroes** [comment]
@cmnrd Maybe to clarify: If you start a language server on port 7670 and then run the first launch configuration that assumes that the language server is available as a jar in the specified directory, you would have a language server running on port 7670 that is not used by the started VS Code extension. So this first setup should only be used if one only edits the VS Code extension but does not plan to change or debug the server.

**Comment by cmnrd** [comment]
Should we add instructions on how to start the correct configuration for the code extension? Or is that already documented elsewhere? I don't know how to use the different launch configurations for VS Code, and it seems not obvious from the description added here. I think some additional information on the entire workflow, including how the VS Code extension talks to the language server that we are debugging, would be really helpful.

**Comment by soerendomroes** [comment]
You can find some general information regarding the Klighd and klighd-vscode extension [here](https://github.com/kieler/KLighD/wiki/Adding-Diagram-Interaction-(on-the-server)), [here](https://github.com/kieler/klighd-vscode/wiki/Diagram-Server-Communication-%E2%80%90-Architectural-Overview), and [here](https://github.com/kieler/klighd-vscode/wiki/Adding-Diagram-Interaction).

These include the general structure but not how launch configurations work (and I think that should be a link the general documentation). Currently, I [tried to add a speaking name](https://github.com/lf-lang/vscode-lingua-franca/pull/157/files) to the launch configurations. What specifically do you want to document regarding the launch configuration (or the debugging setup) and where?
As far as I know, I described it to Peter [here](https://github.com/lf-lang/lingua-franca/discussions/1095) and maybe somewhere else (but I cannot find it) and it seems to be documented [here](https://github.com/lf-lang/vscode-lingua-franca/blob/main/CONTRIBUTING.md).

File: docs/developer/developer-intellij-setup.mdx (hunk 1 above)
Comment (lhstrh) [reply docs/developer/developer-intellij-setup.mdx:74]: Is this ready to get merged? Note that https://github.com/lf-lang/lf-lang.github.io/pull/253 has been merged.

**Comment by cmnrd** [comment]
I was referring to this:
> Per default it uses the port `7670`. This port should be the same as the port specified by the vscode-lingua-franca extension in its [launch configuration](https://github.com/lf-lang/vscode-lingua-franca/blob/38515b4aa0de68383a88a0d4daf56cdb8c5b9b4c/.vscode/launch.json#L25).

which I don't understand what it means. It would be nice to make the description self-contained so that someone, who does not know much about the specifics of the language server and the code extension (like me), has all the information required to actually start a debugging session. I think how to start the language server is well described, but how to interact with it from VS Code is unclear. The description in the extension [CONTRIBUTING.md]() (which apparently is outdated regarding the IntelliJ setup) says 'Then start the extension using the "Launch VS Code Extension (Socket) LF" launch configuration'. But I also don't understand this sentence. How do I start this configuration and using which tool?

**Comment by soerendomroes** [comment]
@cmnrd 
> How do I start this configuration and using which tool?

You start it in VS Code. I do not think that common UI of VS Code should be part of this guide.

VS Code Debug

**Review by petervdonovan — APPROVED** [review]

**Pull request merged**